src/read.c

```diff
@@ -3699,8 +3707,22 @@ static avifResult avifDecoderDataFindAlphaItem(avifDecoderData * data,
         }
     }
     assert(alphaItemCount == colorItemCount);
-    *alphaItem = avifMetaFindItem(colorItem->meta, maxItemID + 1);
-    if (*alphaItem == NULL) {
+    // Figure out the last used itemID.
+    avifResult result = AVIF_RESULT_OK;
+    const uint32_t lastID = colorItem->meta->items.item[colorItem->meta->items.count - 1]->id;
+    if (lastID == UINT32_MAX) {
+        // In the improbable case where the last ID is the maximum one, ids cannot be kept ordered.
+        avifDiagnosticsPrintf(diag,
+                              "Cannot set an itemID for alpha that fits the increasing "
+                              "order as the maximum possible ID is in use.");
+        result = AVIF_RESULT_DECODE_ALPHA_FAILED;
+    } else {
+        *alphaItem = avifMetaFindItem(colorItem->meta, lastID + 1, diag); // Create new empty item.
+        if (*alphaItem == NULL) {
+            result = AVIF_RESULT_OUT_OF_MEMORY;
+        }
+    }
+    if (result != AVIF_RESULT_OK) {
```

Comment:
While verifying the fix for https://crbug.com/oss-fuzz/65657 on the v1.0.x branch, I found an issue with this pull request: line 3728 should be changed from:

```
        return AVIF_RESULT_OUT_OF_MEMORY;
```

to:

```
        return result;
```

This pull request has been reverted because it introduced https://crbug.com/323735410, so this issue is moot now.